certbot-ci/certbot_integration_tests/utils/acme.py

```diff
@@ -0,0 +1,202 @@
+    """Configure and launch the ACME server, Boulder or Pebble"""
+    print('=> Starting {0} instance deployment...'.format(acme_type))
+    instance_path = join(workspace, acme_type)
+    try:
+        # This loads Boulder from git, that includes a docker-compose.yml ready for production.
+        if acme_type == 'boulder':
+            _launch_command(['git', 'clone', 'https://github.com/letsencrypt/boulder',
+                             '--single-branch', '--depth=1', instance_path])
+            # Allow Boulder to ignore usual limit rate policies, useful for tests.
+            os.rename(join(instance_path, 'test/rate-limit-policies-b.yml'),
+                      join(instance_path, 'test/rate-limit-policies.yml'))
+
+        # This configure Pebble using precompiled containers.
+        if acme_type == 'pebble':
+            os.mkdir(instance_path)
+            with open(join(instance_path, 'docker-compose.yml'), 'w') as file_h:
+                file_h.write('''\
+version: '3'
+services:
+  pebble:
+    image: letsencrypt/pebble
+    command: pebble -config /test/config/pebble-config.json {strict} -dnsserver 10.30.50.3:8053
+    ports:
+      - 14000:14000
+    networks:
+      acmenet:
+        ipv4_address: 10.30.50.2
+  challtestsrv:
+    image: letsencrypt/pebble-challtestsrv
+    command: pebble-challtestsrv -defaultIPv6 "" -defaultIPv4 10.30.50.3
+    ports:
+      - 8055:8055
+    networks:
+      acmenet:
+        ipv4_address: 10.30.50.3
+networks:
+  acmenet:
+    ipam:
+      config:
+        - subnet: 10.30.50.0/24
+'''.format(strict='-strict' if acme_option == 'strict' else ''))
+
+        _launch_command(['docker-compose', 'up', '--force-recreate', '-d'], cwd=instance_path)
+
+        # Wait for the ACME CA server to be up.
+        print('=> Waiting for {0} instance to respond...'.format(acme_type))
+        misc.check_until_timeout(acme_xdist['directory_url'])
+
+        # Configure challtestsrv to answer any A record request with ip of the docker host.
+        response = requests.post('http://localhost:{0}/set-default-ipv4'
+                                 .format(acme_xdist['challtestsrv_port']),
+                                 '{{"ip":"{0}.1"}}'.format(acme_xdist['acme_subnet']))
+        response.raise_for_status()
+
+        print('=> Finished {0} instance deployment.'.format(acme_type))
+    except BaseException:
+        print('Error while setting up {0} instance.'.format(acme_type))
+        raise
+
+
+def _prepare_traefik_proxy(workspace, acme_xdist):
+    """Configure and launch Traefik, the HTTP reverse proxy"""
+    print('=> Starting traefik instance deployment...')
+    instance_path = join(workspace, 'traefik')
+    try:
+        os.mkdir(instance_path)
+
+        with open(join(instance_path, 'docker-compose.yml'), 'w') as file_h:
+            file_h.write('''\
+version: '3'
+services:
+  traefik:
+    image: traefik
+    command: --api --rest
+    ports:
+      - "5002:80"
+      - "8056:8080"
```

Comment:
why 8056? can that number be named and put in a variable, especially if we are going to inline config files?

Reply:
Arbitrary number again. What matters is that it is coherent over all parts that rely on its value.

Reply:
On the contrary to the challtestsrv port (8055) that will be used in various places, and so is in `acme_xdist`, this one is used only to the current function. So I put in a local variable.

Reply:
See my general comment below also.